### PR TITLE
feat: add custom theme generation

### PR DESCRIPTION
**Pull request by benjaminpreiss**
- [x] If this is a feature, addition or style change, please **open an issue describing your proposal before opening a pull request.**

### CODE REVIEW

**Comment by benjaminpreiss** [comment]
This closes #261

**Comment by benjaminpreiss** [comment]
@nvkelso what do you think of the changes?

File: styles/src/index.ts [hunk 1]
```diff
@@ -1,50 +1,72 @@
...
 import { LayerSpecification } from "@maplibre/maplibre-gl-style-spec";
 import { labels_layers, nolabels_layers } from "./base_layers";
-import themes, { Theme } from "./themes";
+import themes, { type Theme } from "./themes";
+import j from "jiti";
 
-export default function (source: string, key: string): LayerSpecification[] {
-  const theme = themes[key];
-  return nolabels_layers(source, theme).concat(labels_layers(source, theme));
+export default async function (
```
Comment (benjaminpreiss) [inline styles/src/index.ts:9]: @nvkelso Note, that this has become async!

**Comment by bdon** [comment]
> If this is a feature, addition or style change, please open an issue describing your proposal before opening a pull request.

I'm going to close this PR for now so we can first understand your feature request in the issue, thanks.

**Pull request closed without merging**